### PR TITLE
Fix some combinations of no-* options

### PR DESCRIPTION
**Pull request by mattcaswell**
Fix no-dtls in combination with no-tls.
Fix no-dsa in combination with no-err.

### CODE REVIEW

**Review by hlandau — APPROVED** [review]

File: test/recipes/30-test_evp.t [hunk 1]
```diff
@@ -187,7 +187,7 @@ sub test_errors { # actually tests diagnostics of OSSL_STORE
 }
 
 SKIP: {
-    skip "DSA not disabled", 2 if !disabled("dsa");
+    skip "DSA not disabled", 2 if !disabled("dsa") || disabled("err");
```
Comment (tom-cosgrove-arm) [inline test/recipes/30-test_evp.t:190]: So if both DSA and ERR are disabled, we get the message "DSA not disabled" - won't that be confusing?

File: test/sslapitest.c [hunk 2]
```diff
@@ -1819,6 +1819,9 @@ static int execute_cleanse_plaintext(const SSL_METHOD *smeth,
 
     return testresult;
 }
+#endif /* !defined(OPENSSL_NO_TLS1_2) || !defined(OSSL_NO_USABLE_TLS1_3) \
+        * || !defined(OPENSSL_NO_DTLS)
+        */
```
Comment (tom-cosgrove-arm) [inline test/sslapitest.c:1824]: The comment after the `#endif` is a bit weird - there's a line continuation `\` on the first line, but not on the second. If it's kept to exactly match the `#if` line, maybe the close comment `*/` would be better at the end of line 1823?

File: test/recipes/30-test_evp.t (hunk 1 above)
Comment (mattcaswell) [reply test/recipes/30-test_evp.t:190]: Good point. I updated the error message.

File: test/sslapitest.c (hunk 2 above)
Comment (mattcaswell) [reply test/sslapitest.c:1825]: Closing the comment on the same line would not be consistent with our multi-line comment style. Although, actually the start of this multi-line comment is already not consistent with our style. I fixed it to be consistent with our style. The continuation line was there to match the `#if` - but it doesn't really matter. I removed it.

**Comment by mattcaswell** [comment]
@hlandau - please reconfirm

**Review by hlandau — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by slontis** [comment]
What is the context of these changes.. i.e. Is there existing tests that picked this issue up?

**Comment by mattcaswell** [comment]
> What is the context of these changes.. i.e. Is there existing tests that picked this issue up?

In this comment I talked about a set of options that could be used to try and minimise the size of the openssl executable in order to just be able to use fipsinstall:

https://github.com/openssl/openssl/discussions/21703#discussioncomment-6699586

But when I tried to use them I saw the compile errors I'm fixing here.

**Comment by t8m** [comment]
Merged to master, 3.1, and 3.0 branches. Thank you

**Pull request closed without merging**